parity/cli/usage.rs

```diff
@@ -32,6 +32,17 @@ macro_rules! otry {
 	)
 }
 
+macro_rules! return_if_parse_error {
+	($e:expr) => (
+		match $e {
+			Err(clap_error @ ClapError { kind: ClapErrorKind::ValueValidation, .. }) => {
+				return Err(clap_error);
+			},
+			_ => $e
```

Comment:
it would be better to unwrap the value here rather then calling `ok()` later.

```rust
		match $e {
 			Err(clap_error @ ClapError { kind: ClapErrorKind::ValueValidation, .. }) => {
 				return Err(clap_error);
 			},
 			Ok(e) => e
         }
```

Reply:
What's the reason to ignore errors other than `ClapErrorKind::ValueValidation` btw?

Reply:
@arkpar the clap macro `value_t!` (whose result I feed into the `return_if_parse_error!` macro) yields a `Result` ; the error can be either `ClapErrorKind::ArgumentNotFound` or `ClapErrorKind::ValueValidation`. If it's `ArgumentNotFound` then the value of the argument in the struct should be `None`.

@debris I thought about it but figured it'd be better if the macro didn't convert to Option for separation of concern, but I agree it'd be better for DRY. `$e` can be either `ClapErrorKind::ValueValidation` error, or `ClapErrorKind::ArgumentNotFound` error, or `Ok(...)` ; in the two latter cases this is normal behaviour and we want to convert it to Option; I guess the following would be good?
```
match $e {
 			Err(clap_error @ ClapError { kind: ClapErrorKind::ValueValidation, .. }) => {
 				return Err(clap_error);
 			},
 			_ => $e.ok()
         }
```

Reply:
yes, just describe it in the comment ;)